stdlib/Random/src/RNGs.jl

```diff
@@ -600,6 +600,8 @@ end
 for T in BitInteger_types, R=(1, Inf) # eval because of ambiguity otherwise
     @eval Sampler(::Type{MersenneTwister}, r::UnitRange{$T}, ::Val{$R}) =
         SamplerRangeFast(r)
+    @eval Sampler(::Type{_GLOBAL_RNG}, r::UnitRange{$T}, ::Val{$R}) =
+        SamplerRangeFast(r)
```

Comment:
I guess this mostly solve the problem, but we can imgagine a user specializing similarly on `MersenneTwister` for something else than `UnitRange`, for which this would also be broken with 1.3.

Reply:
Yeah, not sure how to solve that.

Reply:
#33172 ?

Reply:
Yeah, let's go with that